src/EditorFeatures/Core/InlineRename/AbstractInlineRenameUndoManager.cs

```diff
@@ -97,6 +97,12 @@ public void Disconnect()
             this.RedoStack.Clear();
             this.initialState = null;
             this.currentState = null;
+
+            if (_trackedSession is not null)
```

Comment:
This part seems a bit strange to me: I would have expected the only thing that could possibly hold this type would have been the session -- once the session has been let go, this would have fallen away too. Did you see something being held being fixed by this, or you generally noticed this wasn't being updated? Because honestly I'd actually just _delete_ this Disconnect method -- if anything leaks this type directly, we should know that up front it seems?

Reply:
`_trackedSession` only is used to hook up to `ReplacementTextChanged`, which then checks the `currentState`. If `Disconnect` is called prior to the last `ReplacementTextChanged` event we would nullref in  `InlineRenameSession_ReplacementTextChanged`. 

I agree lifetime here is a bit weird. Can pull this out. It was just something I noticed while debugging, did hit a nullref in `InlineRenameSession_ReplacementTextChanged`, so I fixed it. If we're resetting initial and current state, we no longer care about handling `ReplacementTextChanged`. Logically that makes sense to me.

Reply:
(there is a pattern that some parts of Roslyn tried doing which was "if I'm leaked as an object, don't leak other stuff", but it's not a well-liked pattern generally)

Reply:
(that second comment I posted after a call, but GitHub helpfully didn't show your reply yet)

Reply:
> did hit a nullref in InlineRenameSession_ReplacementTextChanged

Is the null ref just because we're nulling everything else out there? Because that'd be a stronger argument for "just delete" if that's practical.

Reply:
It is, and we could do that. I'm happy to do a separate change for that instead of piling into here

Reply:
@ryzngard Sure, that seem good.